cdap-unit-test/src/test/java/co/cask/cdap/security/AuthorizationTest.java

```diff
@@ -1307,15 +1309,41 @@ public void testAddDropPartitions() throws Exception {
     }
   }
 
+  /**
+   * This test is to make sure we do not bypass the authorization check for datasets in system namespace
+   */
+  @Test
+  public void testDeleteSystemDatasets() throws Exception {
+    // create a random user as a remote user since we will set the user back to master user when we access the
```

Comment:
This comment can just be `create a random user and try to delete a system dataset`